spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/config/ShellConfiguration.java

```diff
@@ -42,13 +52,47 @@
 	private static final Logger logger = LoggerFactory.getLogger(ShellConfiguration.class);
 
 	@Bean
-	public TargetHolder targetHolder() {
-		return new TargetHolder();
+	public ApplicationRunner defaultShellRunner(
+			LineReader lineReader,
+			PromptProvider promptProvider,
+			Parser parser,
+			Shell shell) {
+		return new DefaultShellApplicationRunner(lineReader, promptProvider, parser, shell);
+	}
+
+	@Bean
+	public ApplicationRunner initialConnectionApplicationRunner() {
+		return new InitialConnectionApplicationRunner();
 	}
 
 	@Bean
 	public ConsoleUserInput userInput(@Lazy LineReader lineReader) {
 		return new ConsoleUserInput(lineReader);
 	}
 
+	@Order(DefaultShellApplicationRunner.PRECEDENCE - 10)
+	private static class InitialConnectionApplicationRunner implements ApplicationRunner {
+		@Autowired
+		private TargetHolder targetHolder;
+
+		@Autowired
+		private SkipperClientProperties clientProperties;
+
+		@Override
+		public void run(ApplicationArguments args) throws Exception {
+			Target target = new Target(clientProperties.getServerUrl(), clientProperties.getUsername(),
+					clientProperties.getPassword(), clientProperties.isSkipSllValidation());
+
+			// Attempt connection (including against default values) but do not crash the shell on
+			// error
+			try {
+				targetHolder.changeTarget(target, null);
+			}
+			catch (Exception e) {
+				e.printStackTrace();
```

Comment:
We need to decide if we want to keep this (I'd vote yes) or rely on users calling `admin config info` to learn more